### ISSUE TITLE
Make exceptions and warnings part of the public API

### ISSUE THREAD

**Issue by eslavich** [spacetelescope/stdatamodels#13]
CRDS imports `jwst.datamodels.validation.ValidationWarning` so we'll need to make that available somehow.  Maybe add a public `stdatamodels.exceptions` module.

**Comment by emolter** [comment]
I agree `ValidationWarning` makes sense to have as part of the public API.  In `jwst`, it's upgraded to an error in the `pyproject.toml` `filterwarnings`.  `ValidationWarning` is already in the `__all__` of `stdatamodels.jwst.datamodels.validate` but not in the `__all__` of `stdatamodels.validate` where the code actually lives.

I suppose it would be good to at least add a docstring to be picked up by sphinx during the doc build, so that using the Search function of the documentation turns up something useful.

The other potential candidate would be `NoTypeWarning`.  This is in the `__all__` of `stdatamodels.jwst.datamodels.util` where it lives.  The only use in `jwst` is to ignore it in a test.  Not sure if this is used outside of stdatamodels itself
